src/daemon/daemon.go

```diff
@@ -53,6 +53,11 @@ var (
 	ErrDisconnectMaxDefaultConnectionReached                       = errors.New("Maximum number of default connections was reached")
 	// ErrDisconnectMaxOutgoingConnectionsReached is returned when connection pool size is greater than the maximum allowed
 	ErrDisconnectMaxOutgoingConnectionsReached gnet.DisconnectReason = errors.New("Maximum outgoing connections was reached")
+	// ErrDisconnectBlockchainPubkeyNotMatched is returned when the blockchain pubkey in introduction does not match
+	ErrDisconnectBlockchainPubkeyNotMatched gnet.DisconnectReason = errors.New("Blockchain pubkey in Introduction message is not matched ")
+	// ErrDisconnectInvalidExtraData is returned when extra filed can't be parsed as specific data type.
+	// e.g. ExtraData length in IntroductionMessage is not the same as cipher.PubKey
+	ErrDisconnectInvalidExtraData gnet.DisconnectReason = errors.New("Invalid extra data")
```

Comment:
What about using the same error emitted on encoding / decoding error ? In the end incorrect `Extra` bytes would represent an encoding error .

Reply:
What's the encoding / decoding error do you mean? It's not a decode error in either cipher decode nor gnet/message level.

Reply:
IMO this should be detected by as a regular cipher decoding error . Pubkey is encoded as `Extra` bytes though . My comment is because of the fact that not finding a pubkey in `Extra` is caused by wrong encoding by the sending peer , which makes no difference with respect to a decoding error . Indeed if we had `omitempty` for struct types then the binary decoder would have caught this one .

Reply:
This is fine, it lets us disambiguate the problem if it occurs